### PR TITLE
Feature/promptable

### PR DESCRIPTION
**Pull request by worksofliam**
### Changes

This adds the ability to create custom prompt fields on actions. This might be useful when implementing a command but making it nicer on the user to enter specific fields.

```json
{
    "type": "streamfile",
    "extensions": ["rpgle"],
    "name": "Run CRTBNDRPG (inputs)",
    "command": "CRTBNDRPG PGM(${buildlib|Build library|&BUILDLIB}/${objectname|Object Name|&NAME}) SRCSTMF('${sourcePath|Source path|&FULLPATH}') OPTION(*EVENTF) DBGVIEW(*SOURCE) TGTRLS(*CURRENT)"
},
```

image

### Checklist

* [X] have tested my change
* [x] updated relevant documentation
* [X] Remove any/all `console.log`s I added
* [X] eslint is not complaining
* [X] have added myself to the contributors' list in the README

### CODE REVIEW

**Comment by worksofliam** [comment]
@priceaj @barrettotte Would either of you fancy having a review of this?

Thanks!

File: docs/api/custom-ui.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ You can find the source for this API at `src/api/CustomUI.js`.
 
 * `constructor()` creates an instances
 * `addField(Field)` adds a field to the CustomUI
-* `loadPage(context: vscode.ExtensionContext, title: string): Promise<{panel: vscode.WebviewPanel, data: {...}}` is called when you're ready to render the page. The context must come from the extension. The return object contains two pieces of data
+* `loadPage(title: string): Promise<{panel: vscode.WebviewPanel, data: {...}}` is called when you're ready to render the page.
```
Comment (priceaj) [inline docs/api/custom-ui.md:15]: are you missing a '>' here?

**Comment by priceaj** [comment]
Code looks good... but I'm not a first class JS programmer!

Tested by adding an action for those of us who still store their code in source members:

`					{
							"type": "member",
							"extensions": ["RPGLE"],
							"name": "Create Bound RPG Program (CRTBNDRPG) with inputs",
							"command": "CRTBNDRPG PGM(${buildlib|Target library|&OPENLIB}/${objectname|Object Name|&OPENMBR}) SRCFILE(&OPENLIB/&OPENSPF) OPTION(*EVENTF) DBGVIEW(*SOURCE) TGTRLS(*CURRENT)"
						},`

Window pops up, all fields populate correctly, compiling to two separate target libraries works fine

So it's approved from me!

**Comment by worksofliam** [comment]
Thanks @priceaj!

**Pull request merged**